Lib/subprocess.py

```diff
@@ -304,9 +304,9 @@ def call(*popenargs, timeout=None, **kwargs):
     with Popen(*popenargs, **kwargs) as p:
         try:
             return p.wait(timeout=timeout)
-        except:
+        except:  # Including KeyboardInterrupt, wait handled that.
             p.kill()
-            p.wait()
+            # We don't call p.wait() as p.__exit__ does that for us.
```

Comment:
But you're calling `p.kill` *before* `p.__exit__` has a chance of calling `wait(0.25)`... am I missing something?

(same comment for `run()` below)

Reply:
The `p.wait(timeout=timeout)` above where this exception was caught from already did a short wait upon receiving KeyboardInterrupt.

There is always a race condition here though (even before this change):

Popen launches the process in its constructor. KeyboardInterrupt can occur while that is happening, even before it has set self._child_created and exited the constructor.  Or before __enter__ is called, or during __enter__, and within the context manager during all code up and until the try: within wait() itself.  That is impossible to handle and falls back to destructor __del__ or __exit__ based cleanup with no explicit kill() ever happening at all.

The same is just as true within `run()` (which uses communicate(), which similarly handle KeyboardInterrupt).

Reply:
Oops, you're right, sorry... Perhaps spell that "We don't call p.wait() again", then? :-)